testing/mysql/integration_test.go

```diff
@@ -150,6 +150,62 @@ func TestIntegration_MYSQL_SimpleUse(t *testing.T) {
 	checkResults(t, dbPath)
 }
 
+func Integration_MySQLDUMP_SchemaOnly(t *testing.T, dbName, filePrefix, sessionFile, dumpFilePath string) {
+	cmd := exec.Command("bash", "-c", fmt.Sprintf("go run github.com/cloudspannerecosystem/harbourbridge -driver mysqldump -schema-only -dbname %s -prefix %s < %s", dbName, filePrefix, dumpFilePath))
+	var out, stderr bytes.Buffer
+	cmd.Stdout = &out
+	cmd.Stderr = &stderr
+	if err := cmd.Run(); err != nil {
+		fmt.Printf("stdout: %q\n", out.String())
+		fmt.Printf("stderr: %q\n", stderr.String())
+		t.Fatal(err)
+	}
+	if _, err := os.Stat(fmt.Sprintf("%sreport.txt", filePrefix)); os.IsNotExist(err) {
+		t.Fatalf("report file not generated during schema-only test")
+	}
+	if _, err := os.Stat(fmt.Sprintf("%sschema.ddl.txt", filePrefix)); os.IsNotExist(err) {
+		t.Fatalf("legal ddl file not generated during schema-only test")
+	}
+	if _, err := os.Stat(fmt.Sprintf("%sschema.txt", filePrefix)); os.IsNotExist(err) {
+		t.Fatalf("readable schema file not generated during schema-only test")
+	}
+	if _, err := os.Stat(sessionFile); os.IsNotExist(err) {
+		t.Fatalf("session file not generated during schema-only test")
+	}
+}
+
+func Integration_MySQLDUMP_DataOnly(t *testing.T, dbName, dbURI, filePrefix, sessionFile, dumpFilePath string) {
+	cmd := exec.Command("bash", "-c", fmt.Sprintf("go run github.com/cloudspannerecosystem/harbourbridge -driver mysqldump -data-only -instance %s -dbname %s -prefix %s -session %s < %s", instanceID, dbName, filePrefix, sessionFile, dumpFilePath))
+	var out, stderr bytes.Buffer
+	cmd.Stdout = &out
+	cmd.Stderr = &stderr
+	cmd.Env = append(os.Environ(),
+		fmt.Sprintf("GCLOUD_PROJECT=%s", projectID),
+	)
+	if err := cmd.Run(); err != nil {
+		fmt.Printf("stdout: %q\n", out.String())
+		fmt.Printf("stderr: %q\n", stderr.String())
+		t.Fatal(err)
+	}
+	defer dropDatabase(t, dbURI)
+	checkResults(t, dbURI)
+}
+
+func TestIntegration_MySQLDUMP_SchemaOnly_DataOnly(t *testing.T) {
+	tmpdir := prepareIntegrationTest(t)
+	defer os.RemoveAll(tmpdir)
+
+	dbName := "test-individual-flows"
+	dumpFilePath := "../../test_data/mysqldump.test.out"
+	filePrefix := filepath.Join(tmpdir, dbName+".")
+	sessionFile := fmt.Sprintf("%ssession.json", filePrefix)
+	Integration_MySQLDUMP_SchemaOnly(t, dbName, filePrefix, sessionFile, dumpFilePath)
+	// Skip data only test if emulator is not running.
+	onlyRunForEmulatorTest(t)
+	dbURI := fmt.Sprintf("projects/%s/instances/%s/databases/%s", projectID, instanceID, dbName)
+	Integration_MySQLDUMP_DataOnly(t, dbName, dbURI, filePrefix, sessionFile, dumpFilePath)
```

Comment:
Thanks for all the changes in refactoring these tests - consider if following structure makes sense.

// helper functions, only fatal error in case of something unexpected
func runSchemaOnly() {}
func runDataOnly() {}

// Then actual tests
func TestIntegration_MySQLDUMP_SchemaOnly() {
    // setup here ...
    runSchemaOnly()
    // SchemaOnly validations here ...
}

func TestIntegration_MySQLDUMP_DataOnly() {
    // setup here ...
    runSchemaOnly() // no SchemaOnly validations in this case
    runDataOnly()
    // DataOnly validations here ...
}